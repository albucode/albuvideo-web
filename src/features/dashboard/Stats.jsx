import React from "react";
import { Text, Flex, Box } from "@chakra-ui/react";
import styled from "@emotion/styled";

import { IconTag } from "../shared/IconTag";

export const Stats = ({ title, data, icon, inputColor }) => {
  return (
    <Flex>
      <Box>
        <Title>{title}</Title>
        <Data>{data}</Data>
      </Box>
      <IconTag icon={icon} inputColor={inputColor} />
    </Flex>
  );
};

const Title = styled(Text)`
  color: ${(props) => props.theme.colors.grey1};
  font-weight: 400;
  font-size: 18px;
`;

const Data = styled(Text)`
  color: ${(props) => props.theme.colors.black};
  font-weight: 700;
  font-size: 36px;
`;
