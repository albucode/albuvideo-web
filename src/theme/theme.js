import { extendTheme } from "@chakra-ui/react";

const theme = extendTheme({
  fonts: {
    heading: "Open Sans",
    body: "Open Sans",
  },
  colors: {
    table: {
      100: "#F3F2F7",
    },
    lightBlue: "rgb(49, 81, 254)",
    black: "#222222",
    blue: "#3246D2",
    grey1: "#98A1B0",
    grey2: "#F3F2F7",
    grey6: "#EDF1F5",
    grey7: "#E5E5E5",
    magenta: "#E328AF",
    cyan: "#00CFFD",
    red: "#F01010",
  },
});

export default theme;
