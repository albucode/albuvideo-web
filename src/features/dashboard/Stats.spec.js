import React from "react";
import { render, screen } from "@testing-library/react";
import { Stats } from "./Stats";
import Users from "../shared/icons/Users";
import theme from "../../theme/theme";

describe("Stats", () => {
  beforeEach(() => {
    render(
      <Stats
        title={"Total streamed"}
        data={"0h 1"}
        icon={<Users />}
        inputColor={theme.colors.cyan}
      />
    );
  });

  it("displays title", async () => {
    expect(screen.getByTestId("title")).toHaveStyle(
      "font-weight: 400; color: #98A1B0; font-size: 18px"
    );
    expect(screen.getByTestId("title")).toHaveTextContent("Total streamed");
  });

  it("displays data", async () => {
    expect(screen.getByTestId("data")).toHaveStyle(
      "color: #222222; font-weight: 700; font-size: 36px;"
    );
    expect(screen.getByTestId("data")).toHaveTextContent("0h 1");
  });
});
