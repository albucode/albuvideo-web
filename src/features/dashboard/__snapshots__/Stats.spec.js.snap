// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats matches snapshot 1`] = `
<div>
  <div
    class="css-k008qs"
  >
    <div
      class="css-0"
    >
      <p
        class="chakra-text css-1hwa7xk"
        data-testid="title"
      >
        Total streamed
      </p>
      <p
        class="chakra-text css-zobq2s"
        data-testid="data"
      >
        0h 1
      </p>
    </div>
    <span
      class="css-1d0nemg"
      data-testid="square-icon"
    >
      <svg
        class="chakra-icon css-3ufe8c"
        focusable="false"
        viewBox="0 0 48 48"
      >
        <path
          d="M18.412 26.8574C23.0902 26.8574 26.8826 23.065 26.8826 18.3868C26.8826 13.7087 23.0902 9.91626 18.412 9.91626C13.7338 9.91626 9.94141 13.7087 9.94141 18.3868C9.94141 23.065 13.7338 26.8574 18.412 26.8574Z"
          fill="currentColor"
        />
        <path
          d="M24.0589 29.6147H12.7648C6.52726 29.6147 1.4707 34.6712 1.4707 40.9088V45.1441C1.4707 45.9238 2.10274 46.5559 2.88247 46.5559H33.9413C34.7209 46.5559 35.3531 45.9238 35.3531 45.1441V40.9088C35.3531 34.6712 30.2965 29.6147 24.0589 29.6147ZM21.8151 7.61842C26.2845 9.03441 29.5407 13.1409 29.6944 18.0239C29.7006 18.2213 29.8643 18.3842 30.0615 18.3735C34.579 18.1263 38.1672 14.3087 38.0552 9.70531C37.9515 5.44494 34.5071 1.79731 30.2581 1.47022C26.258 1.16225 22.7823 3.63492 21.5785 7.17105C21.5148 7.35802 21.6268 7.5588 21.8151 7.61842Z"
          fill="currentColor"
        />
        <path
          d="M35.2343 21.1384H29.6248C29.4685 21.1384 29.3357 21.2436 29.2944 21.3944C28.7658 23.3257 27.7347 25.0528 26.3437 26.4154C26.1348 26.62 26.2377 26.954 26.5256 27.0046C32.1307 27.9924 36.59 32.2966 37.83 37.808C37.8657 37.9665 38.0025 38.0796 38.165 38.0796H45.1166C45.8931 38.0796 46.5284 37.4584 46.5284 36.6678V32.4325C46.5284 26.2067 41.4742 21.1384 35.2343 21.1384Z"
          fill="currentColor"
        />
      </svg>
    </span>
  </div>
</div>
`;
